src/streamlink/plugins/filmon.py

```diff
@@ -134,7 +134,7 @@ def _get_streams(self):
                 yield stream["quality"], FilmOnHLS(self.session, vod_id=vod_id, quality=stream["quality"])
 
         else:
-            if not channel:
+            if not channel or channel == "sat-1-schweiz":
```

Comment:
I'd create a list of channels that should be handled differently and then check `channel in self._special_case_channels`. In case there are other in the future :)

Reply:
OK, I'll do it.